### PR TITLE
AG-12738 Gauge themes

### PR DESCRIPTION
**Pull request by jacobp100**
https://ag-grid.atlassian.net/browse/AG-12738

### CODE REVIEW

File: packages/ag-charts-community/src/module/optionsModule.ts [hunk 1]
```diff
@@ -100,9 +102,22 @@ export class ChartOptions<T extends AgChartOptions = AgChartOptions> {
...
+            const presetOptions = presetConstructor?.(presetParams, () => this.activeTheme) ?? options;
             this.debug('>>> AgCharts.createOrUpdate() - applying preset', options, presetOptions);
-            options = presetOptions;
+
+            const activeTheme = getChartTheme(options.theme);
+            const presetType = this.optionsType(presetOptions) as any as keyof AgPresetOverrides;
+            const presetTheme = activeTheme.presets[presetType];
+            if (presetTheme != null) {
+                const themedOptions = mergeDefaults(presetParams, presetTheme);
+                options = presetConstructor?.(themedOptions, () => this.activeTheme) ?? themedOptions;
```
Comment (jacobp100) [inline packages/ag-charts-community/src/module/optionsModule.ts]: This is less than ideal - but to determine the theme, we need to know the preset type, so we need to run the preset constructor. Then if it has a theme, we need to run it again with the theme applied

File: packages/ag-charts-community/src/module/optionsModule.ts [hunk 2]
```diff
@@ -100,9 +102,22 @@ export class ChartOptions<T extends AgChartOptions = AgChartOptions> {
...
-            const presetOptions = (PRESETS as any)[this.presetType]?.(options, () => this.activeTheme) ?? options;
+            const presetConstructor: ((options: AgPresetOptions, activeTheme: () => ChartTheme) => T) | undefined = (
+                PRESETS as any
+            )[this.presetType];
+            const presetParams = options as any as AgPresetOptions;
+            const presetOptions = presetConstructor?.(presetParams, () => this.activeTheme) ?? options;
             this.debug('>>> AgCharts.createOrUpdate() - applying preset', options, presetOptions);
-            options = presetOptions;
+
+            const activeTheme = getChartTheme(options.theme);
```
Comment (alantreadway) [inline packages/ag-charts-community/src/module/optionsModule.ts]: Nit: This seems to be duplicating work on later lines, could we move the initialisation of `this.activeTheme` to before this `if` block and only do it once?

File: packages/ag-charts-community/src/module/optionsModule.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/module/optionsModule.ts]: Do we really need to do this - for gauges it looks like `type` is required? For Financial Charts we could fallback to `presetType`?

File: packages/ag-charts-community/src/module/optionsModule.ts (hunk 2 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/module/optionsModule.ts]: Not quite - because financial charts defines `theme` in its returned options

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**